### PR TITLE
Feature/profile form

### PR DESCRIPTION
**Pull request by kncaputo**
## What's this PR do?
- creates `ProfileForm` component w/ inputs for email, shippingAddress, description, and profile picture
- Adds functionality to upload image on `ProfileForm`
- adds route to `ProfileForm` in `src/setup/routes/user.js`
- adds `updateUserInfo` to `web/src/modules/user/api/actions.js`
- added button and `Link` to `Profile` to navigate to `ProfileForm`
- added comments to `Profile` for fields we'll need to add once backend is hooked up

## Relevant tickets
- #78 #55 #54 #53 #52 #51

## Where should the reviewer start?
- review `ProfileForm.js`, `Profile.js`, `actions.js`

## How should this be manually tested?
- test in browser by logging in as a user, clicking `Edit Profile` and uploading a photo (uploading message currently won't timeout)
- navigate back to `Profile`

## Screenshots/GIFs
-
Screen Shot 2021-02-09 at 9 39 14 PM

### CODE REVIEW

File: code/web/src/modules/user/ProfileForm.js [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+import { upload, messageShow, messageHide } from '../common/api/actions'
+
+
+class ProfileForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      userDetails: {
+        id: 0,
+        image: '',
```
Comment (jdcarey128) [inline code/web/src/modules/user/ProfileForm.js:30]: I spoke with Kara about this, but maybe in the future we will want to add the user's current image as well with `this.props.user.details.image`.

File: code/web/src/modules/user/api/actions.js [hunk 2]
```diff
@@ -87,6 +87,16 @@ export function register(userDetails) {
   }
 }
 
+export function updateUserInfo(userDetails) {
+  return dispatch => {
+    return axios.post(routeApi, mutation({
+      operation: 'userUpdate',
+      variables: userDetails,
+      fields: ['user { id, image, email, shippingAddress, description }']
```
Comment (jdcarey128) [inline code/web/src/modules/user/api/actions.js:95]: I think this is formatted correctly. Will likely need to troubleshoot this.

**Review by jdcarey128 — APPROVED** [review]

File: code/web/src/modules/user/ProfileForm.js [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+      userDetails
+    })
+  }
+
+  onUpload = (event) => {
+    this.props.messageShow('Uploading file, please wait...')
+
+    this.setState({
+      isLoading: true
+    })
```
Comment (CalebCyphers) [inline code/web/src/modules/user/ProfileForm.js:53]: I know this is obvious, just want to make a note that we have to remember to set isloading to false once the post request returns.

File: code/web/src/modules/user/ProfileForm.js [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+
+      //     this.props.history.push(admin.productList.path)
+      //   }
+      // })
+      // .catch(error => {
+      //   this.props.messageShow('There was some error. Please try again.')
+
+      //   this.setState({
+      //     isLoading: false
+      //   })
```
Comment (CalebCyphers) [inline code/web/src/modules/user/ProfileForm.js:116]: Oh, here we are, nice

File: code/web/src/modules/user/ProfileForm.js [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+
+                </GridCell>
+              </Grid>
+
+            </form>
+          </GridCell>
+        </Grid>
+
+      </div>
+    )
```
Comment (CalebCyphers) [inline code/web/src/modules/user/ProfileForm.js:222]: Good fix here

**Review by CalebCyphers — APPROVED** [review]

File: code/web/src/modules/user/api/actions.js [hunk 6]
```diff
@@ -87,6 +87,16 @@ export function register(userDetails) {
   }
 }
 
+export function updateUserInfo(userDetails) {
```
Comment (the-color-bliu) [inline code/web/src/modules/user/api/actions.js:90]: Is updateUserInfo the function name we're agreeing on? This will need to be updated in BE if so

**Review by the-color-bliu — APPROVED** [review]
Nice job rebasing, it looks good to me, merge approved

File: code/web/src/modules/user/Profile.js [hunk 7]
```diff
@@ -35,11 +35,18 @@ const Profile = (props) => (
         <H4 style={{ marginBottom: '0.5em' }}>{props.user.details.name}</H4>
 
         <p style={{ color: grey2, marginBottom: '2em' }}>{props.user.details.email}</p>
+        {/* TODO: <p style={{ color: grey2, marginBottom: '2em' }}>{props.user.details.shippingAddress}</p> */}
```
Comment (saraho1123) [inline code/web/src/modules/user/Profile.js:38]: Glad we have these TODOs ready to go!

File: code/web/src/modules/user/ProfileForm.js (hunk 1 above)
Comment (saraho1123) [reply code/web/src/modules/user/ProfileForm.js:30]: That seems right to me as well @jdcarey128 I think we have to wait until we have the connection with BE and FE done!

File: code/web/src/modules/user/ProfileForm.js (hunk 3 above)
Comment (saraho1123) [reply code/web/src/modules/user/ProfileForm.js:53]: Good note Caleb!

**Review by saraho1123 — COMMENTED** [review]
Excited to get this connected with BE! Thanks y'all!

**Pull request merged**